Map plate

Setnea opened the pull request.

Commit:
Adding READ_MAP_PLATE and input validation #62 #63

Commit:
Adding additional functions for map_plate #64

Setnea committed:
[FIX]
Adding value exp_part to map_plate dictionary parser.

Setnea committed:
[MINOR]

[FIX] Renaming task from READ_MAP_PLATE to TASK_READ_MAP_PLATE
[MINOR] TASK_READ_MAP_PLATE will try to get map_plate name defined by user from local dict. The default name is set to "map_plate"

Setnea committed:
[MAPPLATE][FIX] #92

Interchange ID and value keys
Fixing map_plate dict (main well key = well id)

Setnea committed:
[ENH][XML]Command line param parser #86

Parser for additional command line argument allows to add extra
parameter (key, value) to config dictionary during executing program.

Setnea committed:
[LOGS] Changing logs format #21

Setnea committed:
[MINOR] #62 #63

Adding few logs to read_map_plate
Moviing verify_input_file to csv_managment module

Setnea committed:
[FIX] Renaming function

stork119 committed:
Merge pull request #98 from stork119/logs_refactoring

[LOGS] Changing logs format #21

stork119 commented on python/modules/csv_managment.py:
```diff
@@ -5,6 +5,19 @@
 logger = logging.getLogger("CSV managment")
 logger.info("Executing csv (merge) module.")
 
+def verify_input_file(input_file, delimiter, dimensions = 0):
```
@Setnea what does dimensions means ? Add explanation to the function.

stork119 replied:
@Setnea function name is misleading. Because it's not 'verify' any input file, but it checks compliance with the format of one type of mapplate's files. 
Function name  should rather express verification of structure of mapplate file.
Moreover, I'm not sure if it is good practice to open file only for verification of its structure.

Setnea replied:
@stork119 Dimensions inform us about number of rows and cols in csv file. If the value is different then 0, it means that we wish to verify if given files meet given values requirements. Explanation added.

You're right that name migh be missleading, I'm moving this function to map_plate module and renaming it to verify_input_mapplate_file. 
As far as I know you can't extract the number of rows and colls of file without opening it. At least in python.

stork119 commented on python/modules/csv_managment.py:
```diff
@@ -5,6 +5,19 @@
+        x = len(data)
+        y = len(data[0])
+        if dimensions != [x,y]:
+            logger.error("Wrong input file (inproper csv dimensions): %s.", input_file)
```
inproper -> improper

Setnea replied:
Fixed.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -7,30 +7,65 @@
-                new_line.append(j)
-            output_data.append(new_line)
-            num = 1
+def parse_mp(input_path, delimiter):
```
@Setnea it is global function, therefore we should keep a good practice to comment such functions.

Setnea replied:
Comment added.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -7,30 +7,65 @@
+    _parse_params(input_path, delimiter, dim, mp_dict)
+    return mp_dict
+
+def _parse_params(input_path, delimiter, dimensions, mp_dict):
```
@Setnea what does 'params' mean in this case. Please, try to change function name to be more clear or add comment. I think it will be better to create function name in such way that it tells us what does it return, e.g. _get_params_matrix.

Setnea replied:
Comment was added.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -7,30 +7,65 @@
+    for mp_file in param_paths:
+        correctness = CSV_M.verify_input_file(mp_file, delimiter, dimensions)
+        if correctness == False:
+            logger.error("Fatal error occured while trying to read mapplate.")
```
@Setnea we know what is a reason of such error but logs does not inform user about it.

Setnea replied:
@stork119 I can't agree with you. Funcion verify_input_file also provides logs messages and we end up with pretty informative and clear message.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -7,30 +7,65 @@
+            _collect_exp_settings(input_path, mp_file, delimiter, mp_dict)
+
+def _parse_base_params(input_path, delimiter, mp_dict):
+    active_path = FM.path_join(input_path, "args_active.csv")
```
@Setnea what is the difference between 'base' and 'not base' params .

This function is definitely to long. Divide it to smaller, functional parts.

Setnea replied:
@stork119  "Base" params are the params which need to be present in every map_plate: args_active, args_ind, args_name, "not base" params are other information that are facultative and not pre-defined.

I guess I can extract args_active part and verification of args_name and id.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -7,30 +7,65 @@
+            logger.error("Fatal error occured while trying to read mapplate.")
+            exit()
+    if not FM.path_check_existence(name_path):
+        logger.error("Fatal error occured while trying to read mapplate. Cannot parse map_plate names, file doesn't exist: %s", name_path)
```
Part : "Fatal error occured while trying to read mapplate" is hardcoded in many places. Maybe it is possible to extract it. ?

Setnea replied:
Extracted.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -41,95 +76,51 @@ def _getting_paths_mp(input_path):
-                nums = [row, col]
-                active_wells.append(nums)
-    return active_wells
+def _collect_exp_settings(abs_path, mp_file, delimiter, mp_dict):
```
I don't know what this function does.

'abs_path' ?

Setnea replied:
@stork119 This function extract information about optional experiment's parameters (i.e. stimulation.1.1), it parse both name and value of a given parameter for each map_plate well. The name is the same as the name of a similar function from old map_plate module.

abs_path = absolute_path
We're using this shortcut also in other modules (i.e. file_managment).

stork119 commented on python/modules/map_plate.py:
```diff
@@ -41,95 +76,51 @@ def _getting_paths_mp(input_path):
-    return (data[row_nr][col_nr])
-    
-def _getting_column_titles(abs_path, paths):
+def get_param_value(mp_dict, well_name, param): # get values for a given well and parameter
```
Apply changes to functions comments format, according Google Python Style Guide https://google.github.io/styleguide/pyguide.html#Comments

Setnea replied:
Applied.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -41,95 +76,51 @@ def _getting_paths_mp(input_path):
+    value = mp_dict[well_name][param]
+    return value
+
+def get_param_all_values(mp_dict, param): # get all values for a given param from map_plate dictionary
```
'all' in 'get_param_all_values' is redundant -> 'get_param_values'

Setnea replied:
Fixed.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -41,95 +76,51 @@ def _getting_paths_mp(input_path):
+    unique = list(set(values))
+    return unique
+
+def get_all_params_names(mp_dict): # get all key names from map_plate dictionary
```
Same as in 95 line

Setnea replied:
Fixed.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -41,95 +76,51 @@ def _getting_paths_mp(input_path):
+                names = []
+                params_count = tmp
+                for param in mp_dict[well]:
+                    names.append(param)"""
```
Why it is not removed from code ?

Setnea replied:
@stork119 I left if because map_plate is still in development stage and I wasn't sure which assumption will meet all possible cases. If we decide that number of information about wells may somehow differ, we can use that (already written) code.

stork119 commented on python/modules/task.py:
```diff
@@ -304,6 +304,23 @@ def parsing_elements_list(self, dict_local):
         dir_list = FM.dir_get_names(input_path)
         return dir_list, folders_number
 
+class TASK_READ_MAP_PLATE(TASK):
```
Please, add comment about required and optional parameters

Setnea replied:
Added.

stork119 reviewed (changes requested):
Look on the comments in the code and make changes.
Moreover try to keep the rule that lines have only a 80-characters. As it was said before we follow the Foofle Style Guide https://google.github.io/styleguide/pyguide.html

Setnea committed:
[MAPPLATE][FIX]

Fixes following comments in merge map_plate thread.

stork119 committed:
Merge pull request #97 from stork119/master

Merge bugfixes and hotfixes from master (release) to develop

Setnea committed:
[MAPPLATE][FIX]

Fixing read_mapplate, verify map_plate csv and other
functionalities for map_plate data processing.

Setnea committed:
[MAPPLATE] Adding task for applying map_plate #93

Setnea committed:
[FIX]

Setnea committed:
[FIX][MAPPLATE] Spliting position parameter

Spliting well  position parameter to 2 params: position_x and position_y

Setnea committed:
Adding TASK_IF #96

First version of task_if (before xml changes)

stork119 commented on python/modules/map_plate.py:
```diff
@@ -1,69 +1,104 @@
+    Parse all map_plate informations (from csv files) into python dictionary.
+    Each well is represented by dictionary key, 
+    which value is a dictionary of all experimental settings.
+    """
```
@Setnea add more details. Try to present formal and readable structure of mapplate in dictionary. 
function short description
arguments:
- input_path
- delimeter

returns:
- mapplate, ...  structure:
DICT[
---- [ key = args_id
          DICT[ args_name (?)]
  |
  .
  |
---- [key, ...]
]   

This is only example. You can show different representation.

Setnea replied:
@stork119 More formal desciption including all arguments and returns added.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -1,69 +1,104 @@
+    """
+    Params are information about experiment collected in csv files:
+    i.e. cells type, stimulation time.
+    """
```
params - description of experimental procedure in (?) each well; consists for instance: cells type, stimulation time. 
@Setnea

Setnea replied:
It was changed at some point to:

>    Parse information about experiment collected in csv files, i.e.:
>     - cells type
>     - stimulation time
>     - <...>

I can add params definition. However, I prefer to start comment with "parse" due to we describe most of function in such a way (function gets/parse/extracts/does etc)

stork119 commented on python/modules/map_plate.py:
```diff
@@ -1,69 +1,104 @@
+    args_active, args_ind, args_name, 
+    'not base' params are other information that are: 
+    facultative and not pre-defined.
+    """
```
@Setnea try to use nominal sentences and list things, e.g. :
- base params - params ....
- not base params - ...

Setnea replied:
@stork119 fixed

stork119 commented on python/modules/map_plate.py:
```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
+    experiment's parameters (i.e. stimulation.1.1), 
+    it parse both name and value of a given parameter
+     for each map_plate well. 
+    """
```
Change comment as above.

Setnea replied:
Changed.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
-def get_param_value(mp_dict, well_name, param): # get values for a given well and parameter
+def get_param_value(mp_dict, well_name, param):
+    """
+    Function getting values for a given well and parameter.
```
getting ?

Setnea replied:
Gets*, fixed.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
     for well in mp_dict:
-        position = mp_dict[well]["position"]
-        value = (data[position[0]][position[1]])
+        pos_x = int(mp_dict[well]["position_x"])
```
@Setnea does mp_dict keep in postition_x and position_y strings ? Why ?

Setnea replied:
@stork119 Yes, every information in map_plate is saved as a string to allow easy manipulations, easy csv data merging and saving etc. Plus, in fact map_plate is text description of experiment settings. I'm converting those two specific values to (int) because I know what they are and I want to use them as indexes.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
+    column_names = input_data[0]
+    for i in range(len(column_names)):
+        if column_names[i] == compare_column:
+            position = i
```
`+def _prepare_output(input_data, mp_dict, compare_column):
 +    output_data = []
 +    column_names = input_data[0]
 +    for i in range(len(column_names)):
 +        if column_names[i] == compare_column:
 +            position = i`
I'm not sure what happened here ? What does `compare_column` mean?

Setnea replied:
@stork119 
Software is looking for column with given name to extract wells name from it to compare. This function was created for map_plate working after csv_merge (before we discussed problem). I'm gonna change it when new task_parallelize (based on map_plate) will be ready.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
+    return True
+
+
+def _prepare_output(input_data, mp_dict, compare_column):
```
function name _prepare_output is misleading? mapplate output ?
It is important function, add description of arguments.

Setnea replied:
@stork119 Desciption added. Function was renamed to _prepare_mp_output, but maybe _prepare_mp_output_data is more exact (however its pretty long).

stork119 commented on python/modules/map_plate.py:
```diff
@@ -78,49 +113,129 @@ def _get_param_paths(input_path):
+        output_data.append(new_line)
+    return output_data
+
+def apply_mp(input_path, output_path, delimiter, mp_dict, csv_names, compare_column):
```
How do we know which well we analyse now ?
Add description to each function argument.

Setnea replied:
@stork119 Description added. By know it works similar to old map_plate, we know well from well.name column.

stork119 commented on python/modules/task.py:
```diff
@@ -305,22 +305,63 @@ def parsing_elements_list(self, dict_local):
+    - mp_name [name for map_plate output 
+            (variable assigned to dictionary containing all experiment settings), 
+            by default 'map_plate']
+    """
```
👍

stork119 commented on python/modules/task.py:
```diff
@@ -305,22 +305,63 @@ def parsing_elements_list(self, dict_local):
+    - delimiter [delimiter used in csv files, by default ',']
+    - mp_name [name of dicitonary with all collected map_plate info, 
+                by default 'map_plate']
+    """
```
🥇

stork119 commented on python/modules/task.py:
```diff
@@ -305,22 +305,63 @@ def parsing_elements_list(self, dict_local):
+    def execute_specify(self, dict_local):
+        input_path = dict_local["input_path"]
+        output_path = dict_local["input_path"]
+        csv_names = (dict_local["csv_names_list"]).split(",")
```
#102

stork119 commented on python/modules/task.py:
```diff
@@ -305,22 +305,63 @@ def parsing_elements_list(self, dict_local):
+            mp_name = "map_plate"
+        mp_dict = dict_local[mp_name]
+        compare_column = "well.name" # temporary hardcoded [to discuss name and form]
+        map_plate.apply_mp(input_path, output_path, delimiter, mp_dict, csv_names, compare_column)
```
@Setnea this code works now ? There is no "well.name" column in mapplate.

Setnea replied:
@stork119 Actually there is atm. Map_plate works very similar to its old version- it requires csv merge before applying it.

stork119 reviewed (changes requested):
There are some comments in review. Try to create more formal descriptions (using itemize, enumerators) raher than poetic. 

Moreover I still do not understand APPLY function.

stork119 commented on python/modules/task.py:
```diff
@@ -95,10 +95,74 @@ class TASK_QUEUE(TASK):
             dict_local = task.execute(dict_local)
 
+class TASK_IF(TASK):
+
```
Add comment like in TASK_READ_MAP_PLATE

Setnea replied:
Added.

stork119 commented on python/modules/task.py:
```diff
@@ -95,10 +95,74 @@ class TASK_QUEUE(TASK):
+    try:
+        param1 = int(arg1)
+        param2 = int(arg2)
+    except:
```
👎  @Setnea there is a bug. You parse arg to param and further You use arg.

Setnea replied:
Fixed.

stork119 commented on python/modules/task.py:
```diff
@@ -95,10 +95,74 @@ class TASK_QUEUE(TASK):
+        self.task_list = args['task_list']
+
+def execute_specify(self, dict_local):
+    arg1 = dict_local["argument_1"]
```
Add comment that code need to be changed in fututre

Setnea replied:
Commented.

stork119 reviewed (changes requested):
Local functions  must be moved to another module. task.py define some kind of programs api, therefore they cannot share 'engine' functions'

duboviy commented on python/modules/map_plate.py:
```diff
@@ -1,36 +1,106 @@
+    name_path, id_path = _verify_name_and_id(input_path, delimiter, dimensions)
+    name = CSV_M.read_csv(name_path, delimiter)
+    mp_id = CSV_M.read_csv(id_path, delimiter)
+    for row in range(len(active)):
```
Same here

Setnea replied:
@duboviy I changed iteration following your advice. Thank you.

duboviy replied:
@Setnea You are welcome!
I'm glad that my comment was useful for you ;)

duboviy commented on python/modules/map_plate.py:
```diff
@@ -1,36 +1,106 @@
+    name = CSV_M.read_csv(name_path, delimiter)
+    mp_id = CSV_M.read_csv(id_path, delimiter)
+    for row in range(len(active)):
+        for col in range(len(active[row])):
```
Same here

Setnea replied:
Fixed.

duboviy commented on python/modules/map_plate.py:
```diff
@@ -41,95 +111,131 @@ def _getting_paths_mp(input_path):
+def _prepare_output(input_data, mp_dict, compare_column):
+    output_data = []
+    column_names = input_data[0]
+    for i in range(len(column_names)):
```
Same here

Setnea replied:
Fixed.

duboviy commented on python/modules/map_plate.py:
```diff
@@ -41,95 +111,131 @@ def _getting_paths_mp(input_path):
+    output_paths_list = FM.filenames_make_paths_list(output_path, csv_names)
+    logger.debug("Apply map_plate: CSV output files list: "
+                "%s.", output_paths_list)
+    for i in range(len(input_paths_list)):
```
Same here

Setnea replied:
Fixed. However, here, I decided to use zip(input_paths_list, output_paths_list), I belive this solution is good for that case.
Thanks!

duboviy replied:
You are welcome! ;)
zip is great option here, totally agree with you.

duboviy reviewed (changes requested):
- Prefer iteration through object over the loops with the counter. Error with 1 in the index - it's a classic. If the index is required, remember about the enumerate.
```python
# Bad
for i in range(len(xs)) :
    x = xs[i]

# Better
for x in xs:
    ...

# Or
for i, x in enumerate(xs):
    ...
```

Setnea committed:
[MAPPLATE][FIX]

Changes after code review. Adding more formal desriptions, changing
interation to enumerate.

Setnea committed:
Adding new module: flow_control

Adding new module (flow_control), moving TASK_IF functions to it, fixing names bug.

duboviy commented on python/modules/map_plate.py:
```diff
@@ -1,36 +1,106 @@
+    name_path, id_path = _verify_name_and_id(input_path, delimiter, dimensions)
+    name = CSV_M.read_csv(name_path, delimiter)
+    mp_id = CSV_M.read_csv(id_path, delimiter)
+    for row in range(len(active)):
```
You are welcome!
I'm glad that my comment was useful for you ;)

Setnea committed:
[MAPPLATE] Task_parallelize depending on map_plate #95

Adding TASK_PARALLELIZE_MP, parallelize task that depends on map_plate
informations.

Setnea committed:
[MINOR] Updates in tests directory tree

Renaming directory containing test for old software version to IPIQA.2.1. Adding new
directory for current release tests. Update gitignore.

Setnea committed:
[FIX]

Fix in csv_managment write_csv. In case of not existing path to file
software will create needed dirs.

Setnea committed:
[MAPPLATE][FIX] Fixing TASK_APPLY_MAP_PLATE #93

Fixing task to let it works during parallelize based on map_plate.

stork119 commented on python/modules/csv_managment.py:
```diff
@@ -74,6 +74,7 @@ def read_csv(path, mark, dict_local = {}, key_name = ""):
 def write_csv(path, deltimer, data = None, key = ""):
```
@Setnea deltimer -> delimiter

Setnea replied:
I have to change this across all files, but wanted to do whole module at once not just one function.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -227,28 +227,24 @@ def _verify_input_mp_file(input_file, delimiter, dimensions = 0):
     - input_data - data from input csv file
     - mp_dict- map_plate structure containing all information about experiment
-    - compare_column- name of column containing wells' names
+    - mp_key - well id/key to all well parameters in map_plate structure
```
mp_key -- id/key of well in map_plate structure

Setnea replied:
Fixed.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -259,7 +255,6 @@ def apply_mp(input_path, output_path, delimiter, mp_dict,
     - delimiter- separator used in csv files
     - mp_dict- map_plate structure containing all information about experiment
     - csv_names- list of filenames to apply map_plate
-    - compare_column- name of column containing wells' names
```
@Setnea add to comment:
mp_key -- id/key of well in map_plate structure

Setnea replied:
@stork119 added.

stork119 commented on python/modules/flow_control.py:
```diff
@@ -0,0 +1,88 @@
+        arg1 = int(arg1)
+        arg2 = int(arg2)
+    except:
+        logger.error("Cannot compare if given parameter is greater/less than "
```
logger.error(Cannot compare non-numeric arguments: %s, %s", arg1, arg2)

Setnea replied:
Changed.

stork119 commented on python/modules/flow_control.py:
```diff
@@ -0,0 +1,88 @@
+            return True
+    return False
+
+def create_ele_list(input_path, wells_params, used_value):   
```
ele_list ?

Setnea replied:
ele_list -> elements_list  [+ comment]

stork119 commented on python/modules/flow_control.py:
```diff
@@ -0,0 +1,88 @@
+    - wells_params - list of base parameters of chosen wells set
+    - used_value - value defying if given dir name reflects well tag or id
+    Returns
+    - verified_dirst- list of dirs which parameters are present 
```
verified_dirs
I don't understand what this function returned. Comment needs to be updated.
Remove used_value and used only mp_tag for searching well.
Read comment to function get_wells_base_params

Setnea replied:
I changed comments. 
Used_value wasn't remove according to our discussions and previous comments.

stork119 commented on python/modules/flow_control.py:
```diff
@@ -0,0 +1,88 @@
+    """
+    params = []
+    for well in wells:
+        mp_id = prefix + mp_dict[well]["id"] + sufix
```
prefix + mp_dict[well]["id"] + sufix -- this is not part of mapplate so why you name it mp_id ?

It is totally inconsistent naming causing confusion

Setnea replied:
@stork119 Changed according to your advices -> wellname_id, wellname_tag

stork119 commented on python/modules/flow_control.py:
```diff
@@ -0,0 +1,88 @@
+            active_wells.append(well)
+    return active_wells
+
+def get_wells_base_params(mp_dict, wells, prefix, sufix, exp_part):
```
@Setnea meaning of name position in mapplate is as follows :
- if args_name exists then well.name = prefix + args_name + suffix
- else well.name = prefix + args_id + suffix

So, when you create mapplate_structure then add such code: 
if not os.exist("args_name.csv"): 
  mp_dict["name"] = mp_dict["id"]
`if not os.exist("args_name.csv"): 
    mp_dict["name"] = mp_dict["id"]`

So here you should return mp_id (but without any prefixes and sufixes), mp_tag and exp_part

Setnea replied:
@stork119 if you create an issue correlated with your code review comment you should link them to each other to avoid mess #108
We need names with sufixes and prefixes at this point (parallelize_mp).

> 1. I changed names of mp_id and mp_tag.

mp_id and mp_tag -> wellname_id and wellname_tag
> 2. I didn't change functions according to our conversation:
>     - Using names as id if id is missing is already implemented on read_map_plate level
>     - We want to keep the chance to check out both id or tag depends on user preferation (i.e. to allow swaping them during downloading >=2 part experiments.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -38,23 +58,25 @@ def _parse_params(input_path, delimiter, dimensions, mp_dict):
+    1. 'Base' params- params required in each map_plate:
+        - args_active
+        - args_ind
+        - args_name
```
If args_name not exists then args_name = args_ind

Setnea replied:
Good point. Comment added.

Setnea committed:
[FIX]

stork119 commented on python/modules/task.py:
```diff
@@ -253,33 +238,68 @@ def __init__(self, parameters_by_value, parameters_by_name, updates_by_value, up
         processes_number = int(dict_local["number_of_cores"])
         sleep_time = int(dict_local["sleep_time"])
-        new_dirs = dir_list
+        new_elements = elements_list
```
@Setnea new_elements is not redundant variable ?

Setnea replied:
@stork119 No, its not. Elements_list are all collected elements in given loop, while new_elements is a list containing only those which wasn't present before. They're the same only during the first iteration.

Setnea committed:
[FIX] TASK_IF

stork119 commented on python/modules/task.py:
```diff
@@ -253,33 +238,68 @@ def __init__(self, parameters_by_value, parameters_by_name, updates_by_value, up
+        TASK_PARALLELIZE.__init__(self, parameters_by_value, parameters_by_name, updates_by_value, updates_by_name, args)
+        self.config_dict = args['config_dict']
+
+    def parsing_elements_list(self, dict_local): #implementing with tag
```
This function has a lot of repeated parts. Task #105

Setnea replied:
Gonna apply changes from task #105 during/after map_plate merge.

stork119 commented on python/modules/map_plate.py:
```diff
@@ -204,38 +227,47 @@ def _verify_input_mp_file(input_file, delimiter, dimensions = 0):
 
-
-def _prepare_output(input_data, mp_dict, compare_column):
+def _prepare_mp_output(input_data, mp_dict, mp_key):
```
Task #104

Setnea replied:
Discussed in issue topic.

stork119 reviewed (changes requested):
I think that code requires some little changes to be pushed into developed version.

stork119 reviewed (commented):
TASK_IF is ok

Setnea committed:
[FIX] Deltimer -> delimiter #49

Setnea committed:
Merge pull request #106 from stork119/develop

Merging develop changes into map_plate

Setnea committed:
[BUG][FIX] ffc module

Setnea committed:
[TESTS] Adding test for parallelize based on MP

Adding configuration settings for testing new map_plate-> read
map_plate, apply_map_plate and parallelize_mp.

stork119 committed:
[BUG] Test 2017-02-22-t3 updated

Setnea committed:
[BUG] Test3

Uncommenting remove task.

Setnea committed:
[FIX] Changing log msg

Changing log msg following the code review suggestion.

Setnea committed:
Merge branch 'map_plate' of https://github.com/stork119/IPIQA into map_plate

Setnea committed:
[MINOR] #95

Commenting task_parallelize_mp

Setnea committed:
[MINOR] Renaming functions #109

Setnea committed:
Renaming mp params #108

In code and test settings.

Setnea committed:
Test settings changes #108

stork119 commented:
Firstly, you have to merge all previous pull request. Then merge map_plate to develop

Setnea committed:
[FIX] MP code review

Setnea committed:
Updateing test1 and test3 #107

Setnea committed:
[FIX]

uncommenting

Setnea committed:
Interchange id and tag #92 #112

Changing filenames of base map_plate params.
